advanced/src/components/SuperContainer.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react'
+import {
+  NavLink,
+  Link,
+  BrowserRouter as Router,
+  Route,
+  Switch,
+  Redirect,
+} from 'react-router-dom'
+import FeedPage from './FeedPage'
+import DraftsPage from './DraftsPage'
+import CreatePage from './CreatePage'
+import DetailPage from './DetailPage'
+import LoginPage from './LoginPage'
+import SignupPage from './SignupPage'
+import PageNotFound from './PageNotFound'
+import LogoutPage from './LogoutPage'
+import { AUTH_TOKEN } from '../constant'
+import { isTokenExpired } from '../helper/jwtHelper'
+import { withApollo, graphql  } from 'react-apollo'
+import gql from 'graphql-tag'
+
+const ProtectedRoute = ({ component: Component, token, ...rest }) => {
+  return token ? (
+    <Route {...rest} render={matchProps => <Component {...matchProps} />} />
+  ) : (
+    <Redirect to="/login" />
+  )
+}
+
+class SuperContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.refreshTokenFn = this.refreshTokenFn.bind(this)
+
+    this.state = {
+      token: props.token,
+      expireToken: false,
```

Comment:
How come expireToken is never used across the app?

Reply:
its used in 

`  bootStrapData() {
    try {
      const token = localStorage.getItem(AUTH_TOKEN)
      if (token !== null && token !== undefined) {
        const expired = isTokenExpired(token)
        if (!expired) {
          this.setState({ token: token, expireToken: expired })
        } else {
          localStorage.removeItem(AUTH_TOKEN)
          this.setState({ token: null, expireToken: false })
        }
      }
    } catch(e) {
      console.log('')
    }
  }`

Reply:
state.expireToken is not used there either. Nor is it used inside isTokenExpired

```
export function isTokenExpired(token) {
  const date = getTokenExpirationDate(token)
  const offsetSeconds = 0
  if (date === null) {
    return false
  }
  return !(date.valueOf() > new Date().valueOf() + offsetSeconds * 1000)
}
```

Reply:
Enlighten me if I'm wrong.

Reply:
@manjula91 - I believe what @veksen is trying to say is that `expireToken` is set in `bootStrapData()`, however, it is not be referenced/accessed anywhere else in the application. If it isn't being referenced/accessed anywhere then it could be removed.